ios/brave-ios/Sources/PlaylistUI/PlaylistContentView.swift

```diff
@@ -0,0 +1,390 @@
+        }
+      }
+    }
+    .task(priority: .medium) {
+      // FIXME: Will have to adjust this in the future to handle end of TTS
+      for await _ in playerModel.didPlayToEndStream {
+        playNextItem()
+      }
+    }
+    .onAppear {
+      // Make an initial queue assuming nothing selected
+      makeItemQueue(selectedItemID: nil)
+      // Possibly update the selected folder based on the last item played
+      let lastPlayedItem: PlaylistItem? = lastPlayedItemURL.value
+        .map { PlaylistItem.getItems(pageSrc: $0) }?.first
+      if let initialPlaybackInfo,
+        let item = PlaylistItem.getItem(uuid: initialPlaybackInfo.itemUUID)
+      {
+        // If we're coming from a tab with some initial video data update the folder & item
+        seekToInitialTimestamp = initialPlaybackInfo.timestamp
+        selectedItemID = item.id
+        if let folderID = selectedItem?.playlistFolder?.id {
+          selectedFolderID = folderID
+        }
+        // Remake the item queue with the newly selected item
+        makeItemQueue(selectedItemID: selectedItemID)
+      } else {
+        // Just opening playlist without inherited item data, so restore the previously watched item
+        // or just auto-play the first item if the pref is on
+        if let lastPlayedItem {
+          if let lastPlayedItemParentFolder = lastPlayedItem.playlistFolder {
+            selectedFolderID = lastPlayedItemParentFolder.id
+          }
+          selectedItemID = lastPlayedItem.id
+        } else {
+          // Start the first video in the queue if auto-play is enabled
+          if resumeFromLastTimePlayed.value, let selectedItem {
+            seekToInitialTimestamp = selectedItem.lastPlayedOffset
+          }
+          selectedItemID = itemQueue.first
+        }
+      }
+      Task { @MainActor in
+        // If we need to stream the item we need to wait until `loadMediaStreamingAsset` is valid
+        if selectedItem?.cachedData == nil, !loadMediaStreamingAsset.isPrepared {
+          try await Task.sleep(for: .seconds(0.5))
+        }
+        await self.prepareToPlaySelectedItem(
+          initialOffset: seekToInitialTimestamp,
+          playImmediately: initialPlaybackInfo != nil || firstLoadAutoPlay.value
+        )
+      }
+      if selectedItem != nil {
+        selectedDetent = .small
+      } else {
+        selectedDetent = .anchor(.emptyPlaylistContent)
+      }
+    }
+    .onDisappear {
+      if let selectedItem {
+        PlaylistManager.shared.updateLastPlayed(
+          item: .init(item: selectedItem),
+          playTime: playerModel.currentTime
+        )
+      }
+      playerModel.stop()
+    }
+    .alert("New Playlist", isPresented: $isNewPlaylistAlertPresented) {
+      TextField("My New Playlist", text: $newPlaylistName)
+      Button(role: .cancel) {
+        newPlaylistName = ""
+      } label: {
+        Text("Cancel")
+      }
+      .keyboardShortcut(.cancelAction)
+      Button {
```

Comment:
question: So basically we cant disable or enable a button/UI element conditionally in an alert right ? The actions dont work because of it.

Reply:
Yeah the `Create` button will be enabled even if the text box is empty as per comment below this due to SwiftUI bugs